Library/Homebrew/brew.sh

```diff
@@ -600,8 +600,16 @@ if [[ -n "${HOMEBREW_BOTTLE_DEFAULT_DOMAIN}" ]] &&
 then
   unset HOMEBREW_BOTTLE_DOMAIN
 fi
+# HOMEBREW_API_DOMAIN is set from the user environment
+# shellcheck disable=SC2154
+if [[ -n "${HOMEBREW_API_DEFAULT_DOMAIN}" ]] &&
+   [[ "${HOMEBREW_API_DOMAIN}" == "${HOMEBREW_API_DEFAULT_DOMAIN}" ]]
+then
+  unset HOMEBREW_API_DOMAIN
+fi
```

Comment:
The code above this is for compatibility when updating an old-3.1.2 install, due to a now-fixed bug with HOMEBREW_BOTTLE_DOMAIN. This should not be needed for HOMEBREW_API_DOMAIN since the bug should not exist to begin with for that.